src/device/pf_cmrpc.c

```diff
@@ -53,11 +53,44 @@
 #include "pf_block_reader.h"
 #include "pf_block_writer.h"
 
-
+static const char          *rpc_sync_name = "rpc";
 static const pf_uuid_t     implicit_ar = {0,0,0,{0,0,0,0,0,0,0,0}};
 
 /**************** Diagnostic strings *****************************************/
 
+/**
+ * @internal
+ * Display buffer contents
+ *
+ * For example
+ *
+ *    uint8_t mybuffer[12] = {'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L'};
+ *    pf_show_buffer_contents(mybuffer, 12);
+ *
+ * will be displayed as:
+ *
+ * 41 42 43 44 45 46 47 48 49 4A  4B 4C
+ *
+ * Requires 160 columns to be displayed properly.
+ *
+ * @param data            In:   Buffer contents to be displayed
+ * @param size            In:   Buffer size (or smaller that should be displayed)
+ */
+void pf_show_buffer_contents(
```

Comment:
It might be nice to include an ascii representation. Here's a code snippet from another project:

```
static void dwmac1000_pbuf_dump (struct pbuf *p)
{
   int i, j, n;
   char s[80];

   for (i = 0; i < p->len; i += 16)
   {
      n = 0;
      for (j = 0; j < 16 && (i + j) < p->len; j++)
      {
         n += rsnprintf (s + n, sizeof(s) - n, "%02x ",
                         *(uint8_t *)(p->payload + i + j));
      }
      for (; j < 16; j++)
      {
         n += rsnprintf (s + n, sizeof(s) - n, "   ");
      }
      n += rsnprintf (s + n, sizeof(s) - n, "|");
      for (j = 0; j < 16 && (i + j) < p->len; j++)
      {
         uint8_t c = *(uint8_t *)(p->payload + i + j);
         c = (isprint (c)) ? c : '.';
         n += rsnprintf (s + n, sizeof(s) - n, "%c", c);
      }
      n += rsnprintf (s + n, sizeof(s) - n, "|\n");
      DPRINT ("%s", s);
   }
}

```

Reply:
Perhaps a generic pf_show_memory() ?

Reply:
Done